deel/influenciae/influence/arnoldi_influence_calculator.py

```diff
@@ -0,0 +1,330 @@
+# Copyright IRT Antoine de Saint Exupéry et Université Paul Sabatier Toulouse III - All
+# rights reserved. DEEL is a research program operated by IVADO, IRT Saint Exupéry,
+# CRIAQ and ANITI - https://www.deel.ai/
+# =====================================================================================
+"""
+Influence calculator module implementing the approximations in ["Scaling up Influence
+Functions"](https://arxiv.org/pdf/2112.03052.pdf). The Arnoldi algorithm effectively
+reduces the dimension of the problem of computing IHVPs and allows for the calculation
+of influence values on big neural network models.
+"""
+import tensorflow as tf
+
+from ..common import InfluenceModel, BaseInfluenceCalculator, ForwardOverBackwardHVP
+from ..types import Tuple
+
+
+class ArnoldiInfluenceCalculator(BaseInfluenceCalculator):
```

Comment:
I am not very fond of the variables names but if it is mirroring the original implementation then I guess it is ok!